Hmmm, I tried to run 
`python docker_build_test.py kafka/test --image-tag=4.3.0 --image-type=jvm --kafka-url=https://archive.apache.org/dist/kafka/4.3.0/kafka_2.13-4.3.0.tgz` but it failed:
```python
[1/2] STEP 1/8: FROM docker.io/library/eclipse-temurin:21-jre-alpine AS build-jsa
  [1/2] STEP 2/8: USER root
  --> 05cf619f7413
  [1/2] STEP 3/8: ARG kafka_url
  --> 52aea7559fd3
  [1/2] STEP 4/8: COPY jsa_launch /etc/kafka/docker/jsa_launch
  --> f85d5efaec29
  [1/2] STEP 5/8: COPY server.properties /etc/kafka/docker/server.properties
  --> 0aeb529c88b5
  [1/2] STEP 6/8: COPY *kafka.tgz kafka.tgz
  Error: building at STEP "COPY *kafka.tgz kafka.tgz": checking on sources under "/var/tmp/libpod_builder2389327884/build": Rel: can't make  relative to /var/tmp/libpod_builder2389327884/build; copier: stat: ["/*kafka.tgz"]: no such file or directory

  Traceback (most recent call last):
    File "....../docker/common.py", line 46, in build_docker_image_runner
      execute(command.split())
    File "....../docker/common.py", line 27, in execute
      raise SystemError("Failure in executing following command:- ", " ".join(command))
  SystemError: ('Failure in executing following command:- ', 'podman build -f /var/folders/ym/261gbbd9301ck30nz8nlj4b00000gn/T/tmp30p_79xo/jvm/Dockerfile -t kafka/test:4.3.0 --build-arg kafka_url=https://archive.apache.org/dist/kafka/4.3.0/kafka_2.13-4.3.0.tgz
  --build-arg build_date=2026-07-28 --no-cache /var/folders/ym/261gbbd9301ck30nz8nlj4b00000gn/T/tmp30p_79xo/jvm')

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "....../docker/docker_build_test.py", line 82, in <module>
      build_docker_image_runner(f"{container_runtime} build -f $DOCKER_FILE -t {args.image}:{args.tag} --build-arg kafka_url={args.kafka_url} --build-arg build_date={date.today()} --no-cache $DOCKER_DIR", args.image_type)
    File "....//docker/common.py", line 48, in build_docker_image_runner
      raise SystemError(f"{container_runtime.capitalize()} image build failed")
                           ^^^^^^^^^^^^^^^^^
  NameError: name 'container_runtime' is not defined. Did you mean: 'detect_container_runtime'?
```
When I do not set `CONTAINER_RUNTIME` env I think we should always `detect_container_runtime` so avoid such problem.

And also I think there is a second problem (i.e.,  when no `kafka_archive` is provided). I think docker handles it silently a no-match glob. But podman is strictier and would fail with `no such file or directory`:
```python
[1/2] STEP 1/8: FROM docker.io/library/eclipse-temurin:21-jre-alpine AS build-jsa
[1/2] STEP 2/8: USER root
--> 53ebc93d338f
[1/2] STEP 3/8: ARG kafka_url
--> 9bc5b714ea8f
[1/2] STEP 4/8: COPY jsa_launch /etc/kafka/docker/jsa_launch
--> 99e5a53bfd38
[1/2] STEP 5/8: COPY server.properties /etc/kafka/docker/server.properties
--> 988a7e0882df
[1/2] STEP 6/8: COPY *kafka.tgz kafka.tgz
Error: building at STEP "COPY *kafka.tgz kafka.tgz": checking on sources under "/var/tmp/libpod_builder3183074911/build": Rel: can't make  relative to /var/tmp/libpod_builder3183074911/build; copier: stat: ["/*kafka.tgz"]: no such file or directory

Traceback (most recent call last):
  File "..../kafka/docker/common.py", line 49, in build_docker_image_runner
    execute(command.split())
  File "....//kafka/docker/common.py", line 27, in execute
    raise SystemError("Failure in executing following command:- ", " ".join(command))
SystemError: ('Failure in executing following command:- ', 'podman build -f /var/folders/ym/261gbbd9301ck30nz8nlj4b00000gn/T/tmpi7v7rrzj/jvm/Dockerfile -t kafka/test:4.3.0 --build-arg kafka_url=https://archive.apache.org/dist/kafka/4.3.0/kafka_2.13-4.3.0.tgz --build-arg build_date=2026-07-28 --no-cache /var/folders/ym/261gbbd9301ck30nz8nlj4b00000gn/T/tmpi7v7rrzj/jvm')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "..../kafka/docker/docker_build_test.py", line 82, in <module>
    build_docker_image_runner(f"{container_runtime} build -f $DOCKER_FILE -t {args.image}:{args.tag} --build-arg kafka_url={args.kafka_url} --build-arg build_date={date.today()} --no-cache $DOCKER_DIR", args.image_type)
  File "....//kafka/docker/common.py", line 51, in build_docker_image_runner
    raise SystemError(f"{container_runtime.capitalize()} image build failed")
SystemError: Podman image build failed
```

--- a/docker/common.py
+++ b/docker/common.py
@@ -15,6 +15,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from pathlib import Path
 import subprocess
 import tempfile
 import os
@@ -38,14 +39,20 @@ def build_docker_image_runner(command, image_type, kafka_archive=None):
     shutil.copytree(f"{current_dir}/{image_type}", f"{temp_dir_path}/{image_type}", dirs_exist_ok=True)
     shutil.copytree(f"{current_dir}/resources", f"{temp_dir_path}/{image_type}/resources", dirs_exist_ok=True)
     shutil.copy(f"{current_dir}/server.properties", f"{temp_dir_path}/{image_type}")
+
+    kafka_archive_path = Path(temp_dir_path) / image_type / "kafka.tgz"
     if kafka_archive:
-        shutil.copy(kafka_archive, f"{temp_dir_path}/{image_type}/kafka.tgz")
+        shutil.copy(kafka_archive, kafka_archive_path)
+    else:
+        # Podman requires the COPY source to exist before kafka_url is
+        # downloaded by the Dockerfile.
+        kafka_archive_path.touch()
     command = command.replace("$DOCKER_FILE", f"{temp_dir_path}/{image_type}/Dockerfile")
     command = command.replace("$DOCKER_DIR", f"{temp_dir_path}/{image_type}")
     try:
         execute(command.split())
-    except:
-        raise SystemError(f"{container_runtime.capitalize()} image build failed")
+    except Exception as e:
+        raise SystemError("Container image build failed") from e
     finally:
         shutil.rmtree(temp_dir_path)
 

--- a/docker/docker_build_test.py
+++ b/docker/docker_build_test.py
@@ -81,7 +81,7 @@ if __name__ == '__main__':
         if args.kafka_url:
             build_docker_image_runner(f"{container_runtime} build -f $DOCKER_FILE -t {args.image}:{args.tag} --build-arg kafka_url={args.kafka_url} --build-arg build_date={date.today()} --no-cache $DOCKER_DIR", args.image_type)
         elif args.kafka_archive:
-            build_docker_image_runner(f"{container_runtime} build -f $DOCKER_FILE -t {args.image}:{args.tag} --build-arg build_date={date.today()} --no-cache $DOCKER_DIR", args.image_type, args.kafka_archive)
+            build_docker_image_runner(f"{container_runtime} build -f $DOCKER_FILE -t {args.image}:{args.tag} --build-arg kafka_url= --build-arg build_date={date.today()} --no-cache $DOCKER_DIR", args.image_type, args.kafka_archive)
 
     if args.test_only or not (args.build_only or args.test_only):
         run_docker_tests(args.image, args.tag, args.kafka_url, args.image_type, container_runtime)

